core/capabilities/ccip/delegate.go

```diff
@@ -278,17 +278,35 @@ func (d *Delegate) getTransmitterKeys(ctx context.Context, relayIDs []types.Rela
 			return nil, fmt.Errorf("error parsing chain ID, expected big int: %s", relayID.ChainID)
 		}
 
-		ethKeys, err := d.keystore.Eth().EnabledAddressesForChain(ctx, chainID)
-		if err != nil {
-			return nil, fmt.Errorf("error getting enabled addresses for chain: %s %w", chainID.String(), err)
-		}
+		switch relayID.Network {
+		case relay.NetworkEVM:
+			ethKeys, err := d.keystore.Eth().EnabledAddressesForChain(ctx, chainID)
+			if err != nil {
+				return nil, fmt.Errorf("error getting enabled addresses for chain: %s %w", chainID.String(), err)
+			}
 
-		transmitterKeys[relayID] = func() (r []string) {
-			for _, key := range ethKeys {
-				r = append(r, key.Hex())
+			transmitterKeys[relayID] = func() (r []string) {
+				for _, key := range ethKeys {
+					r = append(r, key.Hex())
+				}
+				return
+			}()
+		case relay.NetworkSolana:
+			// Implement EnabledAddressesForChain for Solana as well ?
```

Comment:
This interface has been EVM-only, so I'd keep the comment and maybe tackle this in the future